WordPress/Classes/ViewRelated/Tools/Time Zone/Views/TimeZoneSearchHeaderView.swift

```diff
@@ -0,0 +1,54 @@
+import UIKit
+
+class TimeZoneSearchHeaderView: UIView {
+
+    @IBOutlet private weak var searchWrapperView: SearchWrapperView!
+
+    @IBOutlet private weak var searchWrapperViewHeightConstraint: NSLayoutConstraint!
+
+    @IBOutlet private weak var suggestionLabel: UILabel!
+
+    @IBOutlet private weak var suggestionButton: UIButton!
+
+    /// Callback called when the button is tapped
+    var tapped: (() -> Void)?
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+
+    class func makeFromNib(searchBar: UISearchBar, timezone: String) -> TimeZoneSearchHeaderView {
+        let view = Bundle.main.loadNibNamed(Constants.nibIdentifier,
+                                        owner: self,
+                                        options: nil)?.first as! TimeZoneSearchHeaderView
```

Comment:
The force unwrap here Is less than ideal. It might be unlikely but if, for any reason, that `.xib` file name changes, you would not have any build errors but the app would crash when users try to change the time zone.
Any ideas on how we can improve this?

Reply:
@Gio2018 I've made a change so that the `TimeZoneSearchHeaderView.makeFromNib` class function returns an optional `TimeZoneSearchHeaderView` and the ViewController will use the new `TimeZoneSearchHeaderView` component or fallback to the default `SearchController.SearchBar`.

Couple of comments on to give you an idea of other variations I tried as well:

- I also tried the below which is cleaner in code, but unfortunately the double assignment to tableView.tableHeaderView resulted in a poorer user as the SearchBar sometimes didn't appear which obviously doesn't help our users!

![image](https://user-images.githubusercontent.com/88816724/133484817-d6347b81-f8d2-431e-b379-09d36333dedc.png)


- I also thought about making the makeFromNib return an empty TimeZoneSearchHeaderView as I see that paradigm has been used in another implementation (s/shot below) however I reasoned that having the UI fallback to the default SearchBar is a better user experience.

![image](https://user-images.githubusercontent.com/88816724/133485180-9f60240b-04fa-44c3-8cfb-33ae399362d4.png)


- Re: the force unwrap I had thought that the standard as I had seen it done in these other TableHeaders so was trying to be consistent but this way is definitely safer! 👍 

```
    class func makeFromNib() -> MyProfileHeaderView {
        return Bundle.main.loadNibNamed("MyProfileHeaderView", owner: self, options: nil)?.first as! MyProfileHeaderView
    }

    @objc class func makeFromNib() -> NoteTableHeaderView {
        return Bundle.main.loadNibNamed("NoteTableHeaderView", owner: self, options: nil)?.first as! NoteTableHeaderView
    }

```

Happy to hear your thoughts 😄